db/column_family.cc

```diff
@@ -545,14 +545,34 @@ int GetL0ThresholdSpeedupCompaction(int level0_file_num_compaction_trigger,
   // SanitizeOptions() ensures it.
   assert(level0_file_num_compaction_trigger <= level0_slowdown_writes_trigger);
 
+  if (level0_file_num_compaction_trigger < 0) {
+    return std::numeric_limits<int>::max();
+  }
+
+  const int twice_level0_trigger = level0_file_num_compaction_trigger * 2;
+
+  // overflow protection
+  if (twice_level0_trigger < 0) {
+    return std::numeric_limits<int>::max();
+  }
+
   // 1/4 of the way between L0 compaction trigger threshold and slowdown
   // condition.
+  const int one_fourth_trigger_slowdown =
+      level0_file_num_compaction_trigger +
+      ((level0_slowdown_writes_trigger - level0_file_num_compaction_trigger) /
+       4);
+
+  // overflow protection
+  if (one_fourth_trigger_slowdown < 0) {
```

Comment:
I think one_fourth_trigger_slowdown will never get overflow since level0_slowdown_writes_trigger is its upper bound?